### PR TITLE
Move kSbEventOnScreenKeyboardInvalidTicket, struct SbWindowRect into the on_screen_keyboard.h

### PR DESCRIPTION
**Pull request by iuriionishchenko**
Migrate kSbEventOnScreenKeyboardInvalidTicket, struct SbWindowRect out of starboard/window.h and into the extension header in starboard/extension/on_screen_keyboard.h

b/151173891

Change-Id: Ifd74bf6a2a3781ff3976239314ef0fcfe8c5b487

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]
lgtm if it fully builds. I've triggered the internal CI as well to check on tvOS

File: starboard/extension/on_screen_keyboard.h [hunk 1]
```diff
@@ -25,6 +25,23 @@ extern "C" {
 #define kCobaltExtensionOnScreenKeyboardName \
   "dev.cobalt.extension.OnScreenKeyboard"
 
+#if SB_API_VERSION >= 16
```
[the remark this answers is not in the document]
Comment (iuriionishchenko) [reply starboard/extension/on_screen_keyboard.h:28]: Yes, it was added to prevent compilation errors

**Comment by kaidokert** [comment]
This is currently still failing to build on tvOS

**Comment by iuriionishchenko** [comment]
> This is currently still failing to build on tvOS - looks like missing the include in a few files.

To build on tvOS you should merge this PR and CL - 279300 or use CL - 279280 and CL - 279300

**Comment by kaidokert** [comment]
> > This is currently still failing to build on tvOS - looks like missing the include in a few files.
> 
> To build on tvOS you should merge this PR and CL - 279300 or use CL - 279280 and CL - 279300

Oh i see - so this PR and 279300 need to be submitted together then ?

**Comment by iuriionishchenko** [comment]
> > > This is currently still failing to build on tvOS - looks like missing the include in a few files.
> > 
> > 
> > To build on tvOS you should merge this PR and CL - 279300 or use CL - 279280 and CL - 279300
> 
> Oh i see - so this PR and 279300 need to be submitted together then ?

Yes, they need to be submitted together

**Review by amurovanyi — APPROVED** [review]

**Pull request merged**